plugins/topology/src/utils/topology-utils.ts

```diff
@@ -94,6 +102,45 @@ export const getK8sResources = (
     {},
   );
 
+const isTektonResource = (kind: string) =>
+  [PipelineRunGVK.kind, PipelineGVK.kind].includes(kind);
+
+const getResourceType = (kind: string) => {
+  switch (kind) {
+    case PipelineGVK.kind:
+      return ModelsPlural.pipelines;
+    case PipelineRunGVK.kind:
+      return ModelsPlural.pipelineruns;
+    default:
+      return '';
+  }
+};
+
+export const getTektonResources = (
+  cluster: number,
+  k8sObjects: ObjectsByEntityResponse,
+) =>
+  k8sObjects.items?.[cluster]?.resources?.reduce(
+    (acc: TektonResponseData, res: any) => {
+      if (
+        res.type !== 'customresources' ||
+        (res.type === 'customresources' && res.resources.length === 0)
+      ) {
+        return acc;
+      }
+      const customResKind = res.resources[0].kind;
+      return {
+        ...acc,
+        ...(isTektonResource(customResKind) && {
+          [getResourceType(customResKind)]: {
+            data: res.resources,
+          },
+        }),
+      };
+    },
+    {},
+  );
```

Comment:
Do we need a separate reducer here? custom resources are fetched here https://github.com/janus-idp/backstage-plugins/blob/main/plugins/topology/src/utils/topology-utils.ts#L64-L79 

can we have call in a single place inside one reducer?

Reply:
Wanted to keep it separate for tekton resources for clarity. But if you think it is better to have a single function for both then will update the existing function to get the k8s/openshift resources to include tekton resources as well.